Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace Amazon.Lambda.TestTool.Extensions;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services;
+using Amazon.Lambda.TestTool.Utilities;
+using System.Text;
+using System.Web;
+using static Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyRequest;
+
+/// <summary>
+/// Provides extension methods to translate an <see cref="HttpContext"/> to different types of API Gateway requests.
+/// </summary>
+public static class HttpContextExtensions
+{
+
+    /// <summary>
+    /// Translates an <see cref="HttpContext"/> to an <see cref="APIGatewayHttpApiV2ProxyRequest"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="HttpContext"/> to be translated.</param>
+    /// <param name="apiGatewayRouteConfig">The configuration of the API Gateway route, including the HTTP method, path, and other metadata.</param>
+    /// <returns>An <see cref="APIGatewayHttpApiV2ProxyRequest"/> object representing the translated request.</returns>
+    public static APIGatewayHttpApiV2ProxyRequest ToApiGatewayHttpV2Request(
+        this HttpContext context,
+        ApiGatewayRouteConfig apiGatewayRouteConfig)
+    {
+        var request = context.Request;
+
+        var pathParameters = RouteTemplateUtility.ExtractPathParameters(apiGatewayRouteConfig.Path, request.Path);
+
+        var (headers, _) = HttpRequestUtility.ExtractHeaders(request.Headers);
+        var (queryStringParameters, _) = HttpRequestUtility.ExtractQueryStringParameters(request.Query);
+
+        var httpApiV2ProxyRequest = new APIGatewayHttpApiV2ProxyRequest
+        {
+            RouteKey = $"{request.Method} {apiGatewayRouteConfig.Path}",
+            RawPath = request.Path,
+            RawQueryString = request.QueryString.Value,
+            Cookies = request.Cookies.Select(c => $"{c.Key}={c.Value}").ToArray(),
+            Headers = headers,
+            QueryStringParameters = queryStringParameters,
+            PathParameters = pathParameters ?? new Dictionary<string, string>(),
+            Body = HttpRequestUtility.ReadRequestBody(request),
```

Comment:
Can you take note that when we get these ToAPIGatewayXXX() methods hooked up to the main code path that is asking for the `APIGatewayHttpApiV2ProxyRequest`. When we convert the `APIGatewayHttpApiV2ProxyRequest` to the MemoryStream to Lambda we should check the size of the stream is no more then 6MB. If so throw back the same error that users would get from API Gateway.

Reply:
yeah i will connect with phil on where he wants to do this logic at